### PR TITLE
Add a new menu shell area

### PR DESCRIPTION
**Pull request by jasongrout**
## References

This is an alternative to #9268 that preserves the title at the top of the application.

After playing with both alternatives a bit, I like this PR (combined with #9270) over #9268.

## Code changes

This creates a new 'menu' area of the application that is collapsed into the top area in multiple document mode and pushed just below the top area in single-document mode. Since only the menu is moving down, all other top-level widgets stay at the top of the application. This fixes the jumping noted in #9105.

## User-facing changes

The jumping noted in #9105 is gone.

This PR in action:

sdm2

Stills:
Screenshot_2020-11-03 JupyterLab(6)
Screenshot_2020-11-03 JupyterLab(7)
Screenshot_2020-11-03 JupyterLab(8)
Screenshot_2020-11-03 JupyterLab(9)

## Backwards-incompatible changes

Menus will need to be added to the 'menu' area now, not the 'top' area, to get this automatic single-document behavior.

### CODE REVIEW

**Comment by jtpio** [comment]
> After playing with both alternatives a bit, I like this PR (combined with #9270) over #9268.

Same here :+1:

**Comment by jasongrout** [comment]
> After playing with both alternatives a bit, I like this PR (combined with #9270) over #9268.

Since this PR largely preserves the existing UX, I think it has a lower bar for merging (i.e., it's iterating on the existing design, not asking for consensus on a completely different UX).

**Comment by jtpio** [comment]
Yes it would be useful to have this available to users in the next rc.

Also we might want to mention the `menu` area in the documentation so it's visible to extension authors who could leverage this space in their extension (can be done in a separate PR):

https://jupyterlab.readthedocs.io/en/latest/developer/extension_dev.html#jupyter-front-end-shell

**Review by blink1073 — APPROVED** [review]
Thanks!

**Comment by jasongrout** [comment]
> Also we might want to mention the `menu` area in the documentation so it's visible to extension authors who could leverage this space in their extension (can be done in a separate PR):

Done in this PR now. Thanks for the reminder/suggestion.

**Review by jtpio — APPROVED** [review]
Nice, thanks!

**Comment by jtpio** [comment]
Looks like the PR needs an integrity / lint pass.

**Comment by jasongrout** [comment]
Converting back to draft while I figure out the `application` tests

**Comment by jasongrout** [comment]
Passing application tests now.

**Comment by jasongrout** [comment]
I think this is good to go now, if someone wants to look over the last few commits and press the green button. Then we can close #9268.

File: packages/application-extension/src/index.tsx [hunk 1]
```diff
@@ -942,6 +942,23 @@ const propertyInspector: JupyterFrontEndPlugin<IPropertyInspectorProvider> = {
   }
 };
 
+const JupyterLogo: JupyterFrontEndPlugin<void> = {
+  id: '@jupyterlab/application-extension:logo',
```
Comment (jtpio) [inline packages/application-extension/src/index.tsx:946]: Looks like this plugin is missing the `autostart: true`.

Comment (jtpio) [reply packages/application-extension/src/index.tsx:946]: Fixed in https://github.com/jupyterlab/jupyterlab/pull/9274/commits/a851e913da6e977512f7eaf1f9e4a2317d24f2af

**Review by jtpio — APPROVED** [review]
Thanks!

**Pull request merged**